ui/plugin-system/src/components/PanelSpecEditor/PanelSpecEditor.tsx

```diff
@@ -44,31 +44,21 @@ export function PanelSpecEditor(props: PanelSpecEditorProps) {
     throw new Error(`Missing implementation for panel plugin with kind '${kind}'`);
   }
 
-  const getQueryType = (): string => {
+  const getQueryTypes = (): string[] => {
+    if (plugin.supportedQueryInterfaces && plugin.supportedQueryInterfaces.length > 0) {
```

Comment:
```suggestion
    if (plugin.supportedQueryInterfaces && plugin.supportedQueryInterfaces.length) {
```
I think it should work.